### ISSUE TITLE
admin/functions/download does not respect ARM max response size limits

### ISSUE THREAD

**Issue by michaelpeng36** [Azure/azure-functions-host#8869]
#### Investigative information
The following method in `FunctionsController.cs` https://github.com/Azure/azure-functions-host/blob/7054c51ad9666d6ae0f30da57054d9d8c457789b/src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs#L221 does not seem to respect ARM maximum response size limits, documented here: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#max-response-size.

#### Expected behavior

Calling the endpoint provides a response within the maximum response size limit of 8MB.

#### Actual behavior

Downloads appear to be able to exceed that limit.

**Comment by kashimiz** [comment]
@michaelpeng36 Can you clarify what you mean here? Is the issue that when the `admin/functions/download` endpoint is called through ARM, its response doesn't respect the ARM response size limit?

Please provide an example of a relevant ARM call.

**Comment by michaelpeng36** [comment]
Yes, my understanding is that when the /admin/functions/download endpoint is called through ARM, the response does not respect the ARM response limit. On paper, the limit should be 8MB, but in reality, once the response exceeds 16MB, it starts affecting RP availability. The call itself is `GET/SUBSCRIPTIONS/RESOURCEGROUPS/PROVIDERS/MICROSOFT.WEB/SITES/FUNCTIONS/DOWNLOAD`, with the corresponding URL being of the form `https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{functionAppName}/host/functions/admin/download?api-version=2018-11-01`. Let me know if any more information is needed!

**Comment by fabiocav** [comment]
Tagging @mathewc for discussion/investigation
